src/test/java/uk/nhs/digital/nhsconnect/lab/results/inbound/NhsAckProducerServiceTest.java

```diff
@@ -0,0 +1,237 @@
+package uk.nhs.digital.nhsconnect.lab.results.inbound;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.apache.commons.io.IOUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import uk.nhs.digital.nhsconnect.lab.results.mesh.message.WorkflowId;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Interchange;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.InterchangeHeader;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.MessageHeader;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.InterchangeFactory;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.InterchangeParsingException;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.MessagesParsingException;
+import uk.nhs.digital.nhsconnect.lab.results.sequence.SequenceService;
+import uk.nhs.digital.nhsconnect.lab.results.utils.TimestampService;
+import uk.nhs.digital.nhsconnect.lab.results.inbound.MessageProcessingResult.Success;
+import uk.nhs.digital.nhsconnect.lab.results.inbound.MessageProcessingResult.Error;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+@ExtendWith(MockitoExtension.class)
+public class NhsAckProducerServiceTest {
```

Comment:
JUnit 5 test classes can have default visibility, and this seems to be preferred within the community to distinguish them from JUnit 4- ones. That reason is less important for us since we don't have any JUnit 4 tests anyway, but nevertheless we're trying to keep to the _de facto_ standard.